File: fe/lib/models/ve-xe.dart
import 'user.dart';

class VeXe {
  int? id;
  int? idNv;
  User? nhanVien;
  String? createdDate;
  String? code;
  int? type;
  String? duration;
  String? fullName;
  String? chucVu;
  String? donVi;
  String? sdt;
  String? diaChi;
  int? xe;
  String? moTa;
  String? bienSo;
  String? point;
  int? loaiHinhTt;
  int? statusTt;
  int? status;

  VeXe({
    this.id,
    this.idNv,
    this.createdDate,
    this.nhanVien,
    this.type,
    this.duration,
    this.fullName,
    this.status,
    this.chucVu,
    this.diaChi,
    this.donVi,
    this.loaiHinhTt,
    this.moTa,
    this.bienSo,
    this.point,
    this.sdt,
    this.statusTt,
    this.xe,
  });
  VeXe.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        idNv = json['idNv'],
        createdDate = json['createdDate'],
        code = json['code'],
        nhanVien = User.fromJson(json['nhanVien']),
        type = json['type'],
        duration = json['duration'],
        fullName = json['fullName'],
        chucVu = json['chucVu'],
        diaChi = json['diaChi'],
        donVi = json['donVi'],
        loaiHinhTt = json['loaiHinhTt'],
        moTa = json['moTa'],
        sdt = json['sdt'],
        statusTt = json['statusTt'],
        bienSo = json['bienSo'],
        xe = json['xe'],
        point = json['point'],
        status = json['status'];
  Map<String, dynamic> toJson() => {
        'idNv': idNv,
        'type': type,
        'code': code,
        'duration': duration,
        'fullName': fullName,
        'chucVu': chucVu,
        'diaChi': diaChi,
        'donVi': donVi,
        'loaiHinhTt': loaiHinhTt,
        'moTa': moTa,
        'sdt': sdt,
        'statusTt': statusTt,
        'xe': xe,
        'bienSo': bienSo,
        'point': point,
        'status': status,
      };
}

File: BE/src/main/java/com/haivn/common_api/VeXe.java
package com.haivn.common_api;

import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.DynamicUpdate;
import org.hibernate.annotations.Where;

import javax.persistence.*;
import java.sql.Timestamp;

@Entity
@Table(name = "ve_xe")
@Getter
@Setter
@DynamicUpdate
@Where(clause = "deleted=false")
public class VeXe extends BaseEntity{
    @Column(name = "idNv")
    private Long idNv;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name="idNv",referencedColumnName="id", nullable = false, insertable = false, updatable = false)
    private NguoiDung nhanVien;
    @Column(name = "code")
    private String code;
    @Column(name = "type")
    private Short type;//0: ve thang    1:ve free
    @Column(name = "duration")
    private Timestamp duration;
    @Column(name = "full_name")
    private String fullName;
    @Column(name = "chuc_vu")
    private String chucVu;
    @Column(name = "don_vi")
    private String donVi;
    @Column(name = "sdt")
    private String sdt;
    @Column(name = "dia_chi")
    private String diaChi;
    @Column(name = "xe")
    private Short xe;
    @Column(name = "hang_xe")
    private Short hangXe;
    @Column(name = "bien_so")
    private String bienSo;
    @Column(name = "mo_ta")
    private String moTa;
    @Column(name = "point")
    private String point;
    @Column(name = "loai_hinh_tt")
    private Short loaiHinhTt;
    @Column(name = "status_tt")
    private Short statusTt;
    @Column(name = "status")
    private Short status;
}